File: AutoIt.Foundation/AutoIt.Foundation.Common/LangAnaly/Model/LALR/LALRAction.cs
using Newtonsoft.Json;

namespace AutoIt.Foundation.Common.LangAnaly
{
    public class LALRAction
    {
        public Symbol Symbol { get; set; }
        public ActionType ActionType { get; set; }
        [JsonIgnore]
        public LALRState TargetState { get; set; }
        [JsonIgnore]
        public Produce TargetRule { get; set; }
    }
}

File: AutoIt.Foundation/AutoIt.Foundation.Web.Scripts/CodeEdit/LangAnaly/Model/LALRAction.ts
module CodeEdit.LangAnaly.Model {
    //LALR动作
    export class LALRAction {
        //符号
        Symbol: CodeEdit.LangAnaly.Model.Symbol;
        //动作类型
        ActionType: CodeEdit.LangAnaly.Model.ActionType;
        //目标状态
        TargetState: CodeEdit.LangAnaly.Model.LALRState;
        //目标产生式
        TargetRule: CodeEdit.LangAnaly.Model.Produce;
    }
}